### PR TITLE
Fix of an issue

### PR DESCRIPTION
**Pull request by abhiram555**
fix of #4

### CODE REVIEW

**Comment by abhiram555** [comment]
this pr also has fixes from last closed pr

**Comment by abhiram555** [comment]
fix of #4 and #6

File: src/main/java/de/teddy/advancementhunt/actionbar/ActionbarManager.java [hunk 1]
```diff
@@ -34,7 +34,8 @@ public void run() {
 
 
                 for(Player player : Bukkit.getOnlinePlayers()) {
-                    AdvancementHunt.getInstance().getMessageManager().sendMessageReplace(player, MessageType.TIME_LEFT,"%minutes%",PlaceholderAPI.setPlaceholders(player, "%advancement_time_remaining%"));
+                    // I think this is what you mean about placeholder....
+                    AdvancementHunt.getInstance().getMessageManager().sendMessageReplace(player, MessageType.TIME_LEFT,"%minutes%",PlaceholderAPI.setPlaceholders(player, "%ah_time_remaining%"));
```
Comment (regulad) [inline src/main/java/de/teddy/advancementhunt/actionbar/ActionbarManager.java:38]: Remove `%minutes%` and allow me to put papi placeholders right into the messages,

File: src/main/java/de/teddy/advancementhunt/gamestates/EndingState.java [hunk 2]
```diff
@@ -45,7 +45,7 @@ public void stop() {
                     preparedStatement.setString(2, player.getName());
                     preparedStatement.setInt(3, AdvancementHunt.getInstance().getUtils().getFightUtil().getKills().get(player));
                     preparedStatement.setInt(4, AdvancementHunt.getInstance().getUtils().getFightUtil().getDeaths().get(player));
-                    preparedStatement.setInt(5, AdvancementHunt.getInstance().getUtils().getFightUtil().getLooses().get(player));
+                    preparedStatement.setInt(5, AdvancementHunt.getInstance().getUtils().getFightUtil().getLosses().get(player));
```
Comment (regulad) [inline src/main/java/de/teddy/advancementhunt/gamestates/EndingState.java:48]: Did you check over my changes?

File: src/main/java/de/teddy/advancementhunt/gamestates/IngameState.java [hunk 3]
```diff
@@ -34,7 +37,11 @@ public void start() {
 		}
 		this.advancement_id =  AdvancementHunt.getInstance().getAdvancement_id();
 
-		AdvancementHunt.getInstance().getMessageManager().sendMessageReplace(fleeingPlayer, MessageType.ISFLEEING_PLAYER,"%id%",this.advancement_id);
+		// Fixed Advancement using namspaced key () with multiple placeholders
+		ArrayList<MessageFormat> message_formats = new ArrayList<>();
+		message_formats.add(new MessageFormat("%id%",this.advancement_id));
+		message_formats.add(new MessageFormat("%name%",this.advancement_id.split("/")[1]));
+		AdvancementHunt.getInstance().getMessageManager().sendFormatMessage(fleeingPlayer, MessageType.ISFLEEING_PLAYER,message_formats);
```
Comment (regulad) [inline src/main/java/de/teddy/advancementhunt/gamestates/IngameState.java:44]: I don't want placeholders like this, just add them to the pAPI extension.

File: src/main/java/de/teddy/advancementhunt/listener/PlayerDeathListener.java [hunk 4]
```diff
@@ -77,6 +77,9 @@ private void respawn(Player player) {
             public void run() {
                 player.spigot().respawn();
                 player.sendTitle(new Title("TEST","",1,1,1));
+
+                // Teleport player to spawn if he dies ()
+                player.teleport(Bukkit.getWorld(AdvancementHunt.getInstance().getWorldName()).getSpawnLocation());
```
Comment (regulad) [inline src/main/java/de/teddy/advancementhunt/listener/PlayerDeathListener.java:82]: Don't do this: just use Multiverse to set the spawn the right way.

**Review by regulad — CHANGES_REQUESTED** [review]
Fix these issues and then talk to me again. DO NOT CLOSE THIS PR

File: src/main/java/de/teddy/advancementhunt/actionbar/ActionbarManager.java (hunk 1 above)
Comment (abhiram555) [reply src/main/java/de/teddy/advancementhunt/actionbar/ActionbarManager.java]: Fixed in new commit

**Review by regulad — COMMENTED** [review]
Remember to update `messages.yml`

**Review by regulad — CHANGES_REQUESTED** [review]
`de/teddy/advancementhunt/placeholder/SpigotExpansion` does not have any placeholders that are identical to `%player%` and `%name%`

File: src/main/resources/messages.yml [hunk 5]
```diff
@@ -33,7 +33,7 @@ Messages:
     message: "&cThere are &a%count%&r&c players missing until the start!"
   is_fleeing_player:
     send_type: "TITLE"
-    message: "&cThe hunters are approaching! Get the Advancement %id%!"
+    message: "&cThe hunters are approaching! Get the Advancement %ah_id%!"
```
Comment (regulad) [inline src/main/resources/messages.yml:36]: shouldn't this be `%ah_name%`?

File: src/main/java/de/teddy/advancementhunt/placeholder/SpigotExpansion.java [hunk 6]
```diff
@@ -77,6 +77,13 @@ public String onPlaceholderRequest(Player player, @NotNull String params) {
         }
 
         if(params.equals("id")) {
+            if(AdvancementHunt.getInstance().getAdvancement_id() != null) {
+                return AdvancementHunt.getInstance().getAdvancement_id();
+            }
+        }
+
+        if(params.equals("name"))
```
Comment (regulad) [inline src/main/java/de/teddy/advancementhunt/placeholder/SpigotExpansion.java:85]: looks good

**Comment by regulad** [comment]
`messages.yml` still has `%player%` and `%minutes%`. fix that and this PR will be merged (provided checks pass and I test it)

File: src/main/resources/messages.yml (hunk 5 above)
Comment (abhiram555) [reply src/main/resources/messages.yml:36]: yes, my bad

Comment (abhiram555) [reply src/main/resources/messages.yml:36]: btw that depends on your. if you want advancement with namspaced key or just advancement name

**Comment by abhiram555** [comment]
> `messages.yml` still has `%player%` and `%minutes%`. fix that and this PR will be merged (provided checks pass and I test it)

????

**Comment by abhiram555** [comment]
> `messages.yml` still has `%player%` and `%minutes%`. fix that and this PR will be merged (provided checks pass and I test it)

provide me with change message.yml to what you want then I will make new commit with that

File: src/main/resources/messages.yml [hunk 7]
```diff
@@ -2,41 +2,42 @@
...
-    message: "&cHunter: %player%"
+    message: "Hunter: %ah_player%"
   won:
     send_type: "TITLE"
-    message: "&c%player% won!"
+    message: "%ah_player% won!"
   not_enough_players:
     send_type: "ACTION_BAR"
-    message: "&cThere are &a%count%&r&c players missing until the start!"
+    message: "There are &a%ah_count%&r&c players missing until the start!"
```
Comment (regulad) [inline src/main/resources/messages.yml:34]: This placeholder has not yet been implemented.

File: src/main/resources/messages.yml [hunk 8]
```diff
@@ -2,41 +2,42 @@
...
+    message: "The hunters are approaching! Get the Advancement %ah_name%!"
   Stop_Game:
     send_type: "ACTION_BAR"
-    message: "&cThe game stops in %seconds%..."
+    message: "The game stops in %ah_countdown%..."
   Start_Game:
     send_type: "ACTION_BAR"
-    message: "&cThe game starts in %seconds%..."
\ No newline at end of file
+    message: "The game starts in %ah_countdown%..."
```
Comment (regulad) [inline src/main/resources/messages.yml:43]: This placeholder has not yet been implemented.

File: src/main/resources/messages.yml [hunk 9]
```diff
@@ -2,41 +2,42 @@
 # Configure the messages the plugin displays
 # - send_type
 #   - TITLE: Title Card
+#      - SUBTITLE: Subtitle
```
Comment (regulad) [inline src/main/resources/messages.yml:5]: This feature (part of #6) has not yet been implemented

**Review by regulad — CHANGES_REQUESTED** [review]
@abhiram555 fix this

File: src/main/resources/messages.yml (hunk 7 above)
Comment (abhiram555) [reply src/main/resources/messages.yml:34]: thats not possible, coutdown is just a cache. so adding it papi is not a stable way I suppose

File: src/main/resources/messages.yml (hunk 8 above)
Comment (abhiram555) [reply src/main/resources/messages.yml:43]: thats not possible, coutdown is just a cache. so adding it papi is not a stable way I suppose

Comment (regulad) [reply src/main/resources/messages.yml:43]: undo this change then

File: src/main/resources/messages.yml (hunk 7 above)
Comment (regulad) [reply src/main/resources/messages.yml:34]: undo this change then

**Comment by regulad** [comment]
@abhiram555 https://github.com/squirrelnetmc/AdvancementHunt/pull/18#pullrequestreview-560713494 has still not been fixed

**Comment by abhiram555** [comment]
> @abhiram555 [#18 (review)](https://github.com/squirrelnetmc/AdvancementHunt/pull/18#pullrequestreview-560713494) has still not been fixed

see what I said above

**Comment by regulad** [comment]
> 
> 
> > @abhiram555 [#18 (review)](https://github.com/squirrelnetmc/AdvancementHunt/pull/18#pullrequestreview-560713494) has still not been fixed
> 
> see what I said above

https://github.com/squirrelnetmc/AdvancementHunt/pull/18#discussion_r551034053
https://github.com/squirrelnetmc/AdvancementHunt/pull/18#discussion_r551034046

**Comment by abhiram555** [comment]
alright

**Comment by abhiram555** [comment]
check  now

File: src/main/resources/messages.yml [hunk 10]
```diff
@@ -6,6 +6,7 @@
 #   - CHAT: Chat
 #   - ACTION_BAR: Action Bar
 #   - SUBTITLE: Subtitle
+#   - added this
```
Comment (regulad) [inline src/main/resources/messages.yml:9]: This line isnt needed.

**Review by regulad — CHANGES_REQUESTED** [review]
I see that `%ah_count%` and `%ah_countdown%` still haven't been reverted.

File: src/main/resources/messages.yml (hunk 10 above)
Comment (abhiram555) [reply src/main/resources/messages.yml:9]: well i know i need it for to push it as new commit on discord, you need to change it later

Comment (abhiram555) [reply src/main/resources/messages.yml:9]: well i know but i need that to push it.....,

Comment (abhiram555) [reply src/main/resources/messages.yml:9]: change it your self i suppose.

Comment (regulad) [reply src/main/resources/messages.yml:9]: > well i know but i need that to push it.....,

you still need to fix the other placeholders, `%ah_count%` and `%ah_countdown%`. @abhiram555

Comment (abhiram555) [reply src/main/resources/messages.yml:9]: ???

Comment (abhiram555) [reply src/main/resources/messages.yml:9]: i need to remove it from messages or what?

Comment (regulad) [reply src/main/resources/messages.yml:9]: > 
> 
> i need to remove it from messages or what?

you said that you [cant implement them](https://github.com/squirrelnetmc/AdvancementHunt/pull/18#discussion_r551033506), so change them back to what they were before

**Comment by abhiram555** [comment]
Check now

**Comment by regulad** [comment]
Ok, messages.yml looks fine, What about the queue system in #6? I don't see anything for that.

**Comment by regulad** [comment]
@abhiram555 
# Two Things Left (#6)
* Does the `player` pAPI placeholder work?
1. There is no queue system for messages. Check #6

**Comment by abhiram555** [comment]
Does the player pAPI placeholder work? yes returns player name

**Comment by abhiram555** [comment]
"There is no queue system for messages. Check #6" well how should I implement that?, each messages send separately.

**Comment by regulad** [comment]
> 
> 
> "There is no queue system for messages. Check #6" well how should I implement that?, each messages send separately.

If two messages send at once to ACTION_BAR for example one of them should take priority

**Comment by regulad** [comment]
Maybe add a priority option in `messages.yml`?

**Comment by regulad** [comment]
[This issue](https://github.com/squirrelnetmc/AdvancementHunt/pull/18/files/428eb78b544bfc826436de9d7428105ebb6f6618#r551026607) has also not yet been resolved. @abhiram555

**Comment by abhiram555** [comment]
> [This issue](https://github.com/squirrelnetmc/AdvancementHunt/pull/18/files/428eb78b544bfc826436de9d7428105ebb6f6618#r551026607) has also not yet been resolved. @abhiram555

???

**Comment by regulad** [comment]
# Few Issues
## Fix, then the commission will be closed.
1. Split Storage table into 2: world seed and advancements
1. Allow the plugin to use a DB that is being used by other plugins

**Pull request merged**